src/test/java/unit/GcpDataprocClusterControlled.java

```diff
@@ -0,0 +1,213 @@
+import bio.terra.cli.service.utils.CrlUtils;
+import bio.terra.cloudres.google.dataproc.ClusterName;
+import bio.terra.workspace.model.AccessScope;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnitGcp;
+import harness.utils.GcpDataprocClusterUtils;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matchers;
+import org.json.JSONObject;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra resource` commands that handle controlled GCP clusters. */
+/**
+ * This test only executes on verily environments. Add a tag prefix to prevent it from running in
+ * the main repo
+ */
+@Tag("unit-verily-gcp")
+public class GcpDataprocClusterControlled extends SingleWorkspaceUnitGcp {
+
+  private UFGcpDataprocCluster testCluster;
+  private final String name =
+      "cliTestUserDataprocCluster" + UUID.randomUUID().toString().substring(0, 8);
+  private final String stagingBucketName =
+      "staging-bucket-" + UUID.randomUUID().toString().substring(0, 8);
+  private final String tempBucketName =
+      "temp-bucket-" + UUID.randomUUID().toString().substring(0, 8);
+  private UFGcsBucket stagingBucket;
+  private UFGcsBucket tempBucket;
+
+  // Create controlled cluster resource to use for all tests in this class.
+  @BeforeAll
+  void setupCluster() throws Exception {
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+
+    // Create staging and temp buckets needed by cluster
+    stagingBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "create",
+            "gcs-bucket",
+            "--name=" + stagingBucketName,
+            "--bucket-name=" + stagingBucketName);
+
+    tempBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "create",
+            "gcs-bucket",
+            "--name=" + tempBucketName,
+            "--bucket-name=" + tempBucketName);
+
+    testCluster =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcpDataprocCluster.class,
+            "resource",
+            "create",
+            "gcp-dataproc-cluster",
+            "--name=" + name,
+            "--bucket=" + stagingBucket.bucketName,
+            "--temp-bucket=" + tempBucket.bucketName,
+            "--metadata=foo=bar",
+            "--idle-delete-ttl=1800s");
+
+    GcpDataprocClusterUtils.pollDescribeForClusterState(name, "RUNNING");
+  }
+
+  @Test
+  @DisplayName("list and describe reflect creating a controlled cluster")
+  void listDescribeReflectCreate() throws IOException {
+    workspaceCreator.login();
```

Comment:
i think you should already be logged in since you are extending the singleworkspaceunit